hax/hax/handler.py

```diff
@@ -253,6 +271,9 @@ def update_process_failure(self, planner: WorkPlanner,
                                     HAState(fid=state.fid,
                                             status=current_status))
                         self.process_groups.process_group_unlock(state.fid)
+                        if not is_rc_present:
```

Comment:
Caching, irrespective of the check whether rc is present or not.. to avoid the race window, as suggested by Mandar.
Caching is not done only if (event isn't processed previously by RC or the local node itself.. or incase of node failure)

Reply:
done